pkgs/development/python-modules/scikit-image/default.nix

```diff
@@ -11,20 +12,28 @@
 , pywavelets
 , dask
 , cloudpickle
-, pytest
 , imageio
 , tifffile
+, pytest
 }:
 
 buildPythonPackage rec {
   pname = "scikit-image";
-  version = "0.18.1";
+  version = "0.18.3";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "fbb618ca911867bce45574c1639618cdfb5d94e207432b19bc19563d80d2f171";
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v0.18.3";
+    sha256 = "0a2h3bw5rkk23k4r04qc9maccg00nddssd7lfsps8nhp5agk1vyh";
   };
 
+  # (1) The package has cythonized modules, whose .so libs will appear only in the wheel, i.e. in nix store;
+  # (2) To stop Python from importing the wrong directory, i.e. the one in the build dir, not the one in nix store, `skimage` dir should be (re)moved;
+  # (3) Therefore, tests should be run on the nix-store version.
+  # Because of that, a custom checkPhase is needed along with the patch to add all the data/image files for testing, not only the "legacy" set.
```

Comment:
This doesn't make sense to me... What exactly is wrong?

Reply:
Please see https://discourse.nixos.org/t/cant-import-cythonized-modules-at-checkphase/14207 about the problem with cythonized modules and checks.

Reply:
It would be good to add that link in the comments so that future code readers have a paper trail for this.

Reply:
It seems that a number of existing packages do use a similar trick, but your point is valid, of course.